openfoam-interface/tests/test_foam_interface.py

```diff
@@ -0,0 +1,316 @@
+            for puid in face.points:
+                self.facePoints.append(self._uuidToFoamLabel[puid])
+
+        # make points coordinate list
+        self.pointCoordinates = []
+        for point in self.mesh.iter_points():
+            for coord in point.coordinates:
+                self.pointCoordinates.append(coord)
+
+        # make compressed list of cells points
+        self.cellPoints = []
+        for cell in self.mesh.iter_cells():
+            self.cellPoints.append(len(cell.points))
+            for puid in cell.points:
+                self.cellPoints.append(self._uuidToFoamLabel[puid])
+
+        # make patch information
+        self.patchNames = []
+        self.patchFaces = []
+        for patchName in patchNameFacesMap:
+            self.patchNames.append(patchName)
+            self.patchFaces.append(len(patchNameFacesMap[patchName]))
+            for face in patchNameFacesMap[patchName]:
+                self.patchFaces.append(face)
+
+        if not self.patchNames:
+            error_str = 'Could not initialize with mesh  {}. '
+            error_str += 'Mesh has not boundary face definitions.'
+            raise ValueError(error_str.format(self.mesh.name))
+
+        # this to have controlDict file for mesh definition
+        foam_files.write_default_files(
+            self.path, 'simpleFoam', self.time, False)
+
+        foamface.init(self.name, os.path.abspath(os.path.join(self.path,
+                                                              os.pardir)))
+
+    def tearDown(self):
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+
+    def test_add_mesh(self):
+        """Test addMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        pointCoordinates = foamface.getAllPointCoordinates(self.name)
+        self.assertEqual(self.pointCoordinates, pointCoordinates)
+        cellPoints = foamface.getAllCellPoints(self.name)
+        self.assertEqual(set(self.cellPoints), set(cellPoints))
+        facePoints = foamface.getAllFacePoints(self.name)
+        self.assertEqual(self.facePoints, facePoints)
+        patchNames = foamface.getBoundaryPatchNames(self.name)
+        self.assertEqual(self.patchNames, patchNames)
+        patchFaces = foamface.getBoundaryPatchFaces(self.name)
+        self.assertEqual(self.patchFaces, patchFaces)
+
+    def test_write_mesh(self):
+        """Test writeMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        foamface.writeMesh(self.name)
+        meshpath = os.path.join(self.test_dir, self.name,
+                                'constant', 'polyMesh')
+        self.assertEqual(os.path.exists(
```

Comment:
`suggestion`: It preferable to use `self.assertTrue` for checks that assert value == True (see table  https://docs.python.org/2/library/unittest.html#assert-methods), In general the preferable assert method is the one that is as specific as possible on what we aim to assert.